src/Hamiltonians/excitations.jl

```diff
@@ -139,6 +135,27 @@ function momentum_transfer_diagonal(map::BoseOccupiedModeMap)
     end
     return float(onproduct)
 end
+
+"""
+    extended_momentum_transfer_diagonal(map, M)
+
+The diagonal part of nearest neighbour [`momentum_transfer_excitation`](@ref).
+"""
+
```

Comment:
Please remove the empty line 144! Otherwise the docstring won't be found.

It would also be nice to have a bit more verbose explanation of what the arguments mean and what is returned.